core/schema/git.go

```diff
@@ -104,6 +106,13 @@ func (s *gitSchema) git(ctx context.Context, parent *core.Query, args gitArgs) (
 	}, nil
 }
 
+func (s *gitSchema) head(ctx context.Context, parent *core.GitRepository, args struct{}) (*core.GitRef, error) {
+	return &core.GitRef{
+		Query: parent.Query,
+		Repo:  parent,
```

Comment:
How does this work? Is it actually OK to leave `Ref` as `""`? :thinking:

Reply:
https://github.com/dagger/dagger/blob/64e6ab0dab0da4d595bddc68a8b493e6286309f0/engine/sources/gitdns/source.go#L495-L501

:tada: buildkit works this out (as well as our copy-paste job here)